File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& nums) {
        int ans = 0;
        int maxEl = nums[nums.size()-1];
        for(int i = nums.size()-2; i>=0; i--){
            ans = max(ans, maxEl - nums[i]);
            if(maxEl < nums[i]) maxEl = nums[i];
        }
        return ans;
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        def maxEl(start:int, end:int, nums:List[int]) -> int:
            maxEl = nums[start]
            while start < end:
                maxEl = max(maxEl, nums[start])
                start += 1
            return maxEl
        
        i,j = 0, k
        ans = []
        while j <= len(nums):
            maxi = maxEl(i,j,nums)
            ans.append(maxi)
            i += 1
            j += 1
        return ans